Data/ByteString.hs

```diff
@@ -332,6 +332,10 @@ unpackFoldr bs k z = foldr k z bs
 -- that allow arbitrary bytes in their paths. This conversion
 -- function does the same thing that `System.IO.openFile` would
 -- do when decoding the 'FilePath'.
+--
+-- This function is in 'IO' because the file system encoding can be
+-- changed. If you plan on keeping the encoding constant, then you
+-- can use 'unsafePerformIO'.
```

Comment:
I've tried to polish the second sentence. Not sure whether it's actually better.

```suggestion
-- changed. If the encoding can be assumed to be constant in your
-- use case, you may invoke this function via 'unsafePerformIO'.
```

Reply:
Agreed. That does sound less awkward.